runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/environment/DockerEnvironmentFactory.java

```diff
@@ -168,4 +173,21 @@ public RemoteEnvironment createEnvironment(Environment environment) throws Excep
 
     return DockerContainerEnvironment.create(docker, environment, containerId, instructionHandler);
   }
+
+  private List<String> gcsCredentialArgs() {
+    String dockerGcloudConfig = "/root/lconfig/gcloud";
```

Comment:
`lconfig` -> `.config`?

Reply:
What happens if .config/gcloud already exists?

Reply:
I thought `lconfig` is a typo - it's not a folder recognized by gcloud, but `.config` is. I'm not sure what happens if it already exists, but `lconfig` doesn't look right.

Reply:
Yes, the environment variable is set below. 

On the other hand, if we're pretty confident about non-collision, maybe we should just hook it up into the "extra mount points" code where it would just be a `"~/.config/gcloud" -> "/root/.config/gcloud"` entry.

Reply:
OK, looks like this may actually be respected by the Java auth library: https://github.com/google/google-auth-library-java/pull/63. If we've tested this and are pretty confident, this should work. It would be good to at least include an inline comment about why we're not using the standard `.config` directory.

Reply:
Given this discussion, I've simplified to just use `.config/gcloud`.